indico/modules/categories/util_test.py

```diff
@@ -0,0 +1,27 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2021 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+import pytest
+
+from indico.modules.categories.util import can_create_unlisted_events
+from indico.modules.events.settings import unlisted_events_settings
+
+
+@pytest.mark.usefixtures('db')
+def test_can_create_unlisted_events(dummy_user, create_user):
+    assert not can_create_unlisted_events(None)
+    unlisted_events_settings.set('enabled', False)
+    unlisted_events_settings.set('restricted', False)
+    assert not can_create_unlisted_events(dummy_user)
```

Comment:
I'd check for the `admin_user` here as well (especially since the initial version of the PR did not disable it for admin users even if the feature was disabled)